KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,327 @@
+            del khyper_header_chap[a]
+            del k_hyper_sub_chap[a]
+        else:
+            a += 1
+
+    #global sorter_check
+    chap9 = 0
+
+    if sorter_check[sortloc][0] == 0:
+        sorter_check[sortloc][0] += 1
+        chap9 = 1
+    else:
+        offset = sorter_check[sortloc][1]
+        # if special_input == 1:
+        #     offset = 8
+
+    if special_input in (2, 3):
+        name_chap = 'hypergeometric representation'
+
+    # Uses of numbers like 12, or 9 are for specific lengths of strings (like \\subsection{) where the
+    # section does not change
+    for d in range(0, len(tempref)):  # check every section and subsection line
+        item = tempref[d]
+        line = str(chap[item])
+        if "\\section{" in line or "\\subsection{" in line:
+            if "\\subsection{" in line:
+                temp = line[12:line.find("}", 7)]
+            else:
+                temp = line[9:line.find("}", 7)]
+
+        if name_chap in line.lower():
+            if special_input in (0, 2, 3, 4) or special_input == 1 and "orthogonality relation" not in line:
+                hyper_subs_chap.append([tempref[d + 1]])  # appends the index for the line before following subsection
+                hyper_headers_chap.append(temp)  # appends the name of the section the hypergeo subsection is in
+
+                if temp in khyper_header_chap:
+                    try:
+                        chap[tempref[d + 1] - 1] += "\paragraph{\\bf KLS Addendum: " + word + "}" + \
+                                                    k_hyper_sub_chap[k_hyp_index_iii + offset]
+                        if chap9 == 1:
+                            sorter_check[sortloc][1] += 1
+                        k_hyp_index_iii += 1
+                    except IndexError:
+                        print("Warning! Code has found an error involving section finding for '" + name_chap + "'.")
+
+
+    if len(hyper_headers_chap) != 0:
+        # print "Stuff for checking"
+        # print k_hyper_sub_chap
+        # print khyper_header_chap
+        # print hyper_headers_chap
+        # print word
+        sortmatch_2.append(k_hyper_sub_chap)
+    if "generating function" in name_chap:
+        print "Stuff for checking"
+        print k_hyper_sub_chap
+        print khyper_header_chap
+        print hyper_headers_chap
+    return chap
+
+
+def cut_words(word_to_find, word_to_search_in):
+    """
+    This function checks through the outputs of later sections and removes duplicates, so that the output file does
+    not have duplicates.
+
+    IN DEVELOPMENT
+
+    :param word_to_find: Word that is being found
+    :param word_to_search_in: The big word that is being searched
+    :return: The big word without the word that was searched for
+    """
+    a = word_to_find
+    b = word_to_search_in
+    precheck = 1
+    if a in b:
+        if "\\paragraph{\\bf KLS Addendum: " in b or "\\subsubsection*{\\bf KLS Addendum: " in b:
+            while True:
```

Comment:
`while not [condition]` instead of `while True` and breaking to make it more readable